plugins/inputs/conntrack/conntrack.go

```diff
@@ -57,9 +57,33 @@ func (*Conntrack) SampleConfig() string {
 	return sampleConfig
 }
 
-func (c *Conntrack) Gather(acc telegraf.Accumulator) error {
+func (c *Conntrack) Init() error {
+	var all bool
+	var perCPU bool
+
 	c.setDefaults()
 
+	for _, collect := range c.Collect {
+		switch collect {
+		case "all":
+			all = true
+		case "percpu":
+			perCPU = true
+		default:
+			return fmt.Errorf("collect %s is not supported", collect)
+		}
+	}
+
+	if all || perCPU {
+		if c.ps == nil {
```

Comment:
I think `c.ps` is _always_ `nil` here due to the fact that `Init()` is only called once. So simply remove this check?

Reply:
Thanks. No, we can't. In the tests [we pass](https://github.com/deric/telegraf/blob/conn_segfault/plugins/inputs/conntrack/conntrack_test.go#L129)  the mock instance, thus the `c.ps` won't be `nil`. If we remove the check the tests will collect conntrack data from the CI instance.

Reply:
Hmmm so why not override `c.ps` _after_ calling `Init()`? I really hate to change "main" code just to accommodate tests... If there really is no other way I don't want to hold up this PR but this really hurts my eyes. ;-)

Reply:
I guess this is the cost of lazy initialization. I moved the initialization, but `ps` will be initialized even in cases when it's not needed.

Reply:
I can live with that as it is only in tests...